File: frontend/src/stores/skills.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import { skillsApi } from '@/api'

export interface Skill {
  id: string
  name: string
  enabled: boolean
  config: Record<string, any>
}

export interface ImportSource {
  source: string
  path: string
  type: string
  skills: { id: string; name: string }[]
}

export const useSkillsStore = defineStore('skills', () => {
  const skills = ref<Skill[]>([])
  const importSources = ref<ImportSource[]>([])
  const loading = ref(false)

  async function fetchSkills() {
    loading.value = true
    try {
      const res = await skillsApi.list()
      skills.value = res.data.skills || []
    } finally {
      loading.value = false
    }
  }

  async function fetchImportSources() {
    const res = await skillsApi.listSources()
    importSources.value = res.data.sources || []
  }

  async function toggleSkill(id: string, enabled: boolean) {
    await skillsApi.toggle(id, enabled)
    const skill = skills.value.find(s => s.id === id)
    if (skill) skill.enabled = enabled
  }

  async function updateSkill(id: string, config: Record<string, any>) {
    const skill = skills.value.find(s => s.id === id)
    if (!skill) return
    await skillsApi.update(id, { ...skill, config })
    skill.config = config
  }

  async function importSkill(source: string) {
    const res = await skillsApi.import(source)
    await fetchSkills()
    await fetchImportSources()
    return res.data
  }

  async function unimportSkill(source: string) {
    const res = await skillsApi.unimport(source)
    await fetchSkills()
    await fetchImportSources()
    return res.data
  }

  return { skills, importSources, loading, fetchSkills, fetchImportSources, toggleSkill, updateSkill, importSkill, unimportSkill }
})

File: backend/routers/skills.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from config import load_json, save_json
from services.import_skill import SkillImportService

router = APIRouter()
import_service = SkillImportService()

class SkillConfig(BaseModel):
    id: str
    name: str
    enabled: bool = True
    config: dict = {}

class ImportRequest(BaseModel):
    source: str  # URL or local absolute path

@router.get("/")
def list_skills():
    data = load_json("skills.json")
    return data

@router.post("/import")
async def import_skill(request: ImportRequest):
    """Import a skill from URL (cloud) or local path (symlink or copy)."""
    result = import_service.import_skill(request.source)
    if not result.get("success"):
        raise HTTPException(status_code=400, detail=result.get("error", "Import failed"))
    return result

@router.get("/imported")
def list_imported_skills():
    """List all locally imported/skilled skills."""
    return {"skills": import_service.list_imported_skills()}

@router.get("/import-sources")
def list_import_sources():
    """List all import sources (paths/URLs) with their managed skills."""
    return {"sources": import_service.list_import_sources()}

@router.delete("/import")
async def unimport_skill(source: str):
    """Remove an imported skill source and all its skills from management."""
    result = import_service.unimport_skill(source)
    if not result.get("success"):
        raise HTTPException(status_code=400, detail=result.get("error", "Unimport failed"))
    return result

@router.put("/{skill_id}")
def update_skill(skill_id: str, skill: SkillConfig):
    data = load_json("skills.json")
    for i, s in enumerate(data.get("skills", [])):
        if s["id"] == skill_id:
            data["skills"][i] = skill.model_dump()
            save_json("skills.json", data)
            return {"success": True}
    raise HTTPException(status_code=404, detail="Skill not found")

@router.put("/{skill_id}/enabled")
def toggle_skill(skill_id: str, enabled: bool):
    data = load_json("skills.json")
    for i, s in enumerate(data.get("skills", [])):
        if s["id"] == skill_id:
            data["skills"][i]["enabled"] = enabled
            save_json("skills.json", data)
            return {"success": True}
    raise HTTPException(status_code=404, detail="Skill not found")
